<?xml version="1.0"?>
<launch>
  <!-- Launch the OpenNI drivers -->
        <!--param name="/camera/driver/color_mode" value="1" /-->
	<arg name="camera"      default="camera"/>
	<arg name="publish_tf"  default="false"/>
	<arg name="depth_registration"              default="true"/>
	<arg     if="$(arg depth_registration)" name="depth" value="depth_registered" />
	<include file="$(find openni2_launch)/launch/openni2.launch" >
		<arg name="publish_tf" value="$(arg publish_tf)"/>
		    <arg name="rgb_processing" value="true" />
			<arg name="depth_registration" value="true" />
			<arg name="depth_processing" value="true" />
	</include>

  <!-- Run the depthimage_to_laserscan node -->
<!--
  <node pkg="depthimage_to_laserscan" type="depthimage_to_laserscan" name="depthimage_to_laserscan" output="screen">
    <remap from="image" to="/camera/depth/image_raw" />
    <remap from="camera_info" to="/camera/depth/camera_info" />
    <remap from="scan" to="/scan" />
	</node>
-->

	<arg name="scan_topic" default="scan"/>
    <node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan" args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet $(arg camera)/$(arg camera)_nodelet_manager">
      <!-- Pixel rows to use to generate the laserscan. For each column, the scan will
           return the minimum value for those pixels centered vertically in the image. -->
      <param name="scan_height" value="10"/>
      <!--param name="scan_time" value="0.2" /-->
      <param name="output_frame_id" value="/$(arg camera)_depth_frame"/>
      <param name="range_min" value="0.45"/>
      <param name="range_max" value="8.0" />
      <param name="scan_time" value="0.07" />
      <remap from="image" to="$(arg camera)/$(arg depth)/image_raw"/>
      <remap from="scan" to="$(arg scan_topic)"/>

      <!-- Somehow topics here get prefixed by "$(arg camera)" when not inside an app namespace,
           so in this case "$(arg scan_topic)" must provide an absolute topic name (issue #88).
           Probably is a bug in the nodelet manager: https://github.com/ros/nodelet_core/issues/7 -->
      <remap from="$(arg camera)/image" to="$(arg camera)/$(arg depth)/image_raw"/>
      <remap from="$(arg camera)/scan" to="$(arg scan_topic)"/>
    </node>

</launch>
